From: Shiji Yang <yangshiji66@outlook.com>
Date: Thu, 1 May 2025 01:59:06 +0800
Subject: [PATCH] irqchip/ath79-misc: fix missing prototypes warnings

ath79_misc_irq_init() was defined but unused since commit
51fa4f8912c0 ("MIPS: ath79: drop legacy IRQ code"), so it's time
to drop it. This patch also adds the missing header "asm/time.h"
which defines the prototype of get_c0_perfcount_int().

Fix the following warnings:

drivers/irqchip/irq-ath79-misc.c:26:5: error: no previous prototype for 'get_c0_perfcount_int' [-Werror=missing-prototypes]
   26 | int get_c0_perfcount_int(void)
      |     ^~~~~~~~~~~~~~~~~~~~
drivers/irqchip/irq-ath79-misc.c:181:13: error: no previous prototype for 'ath79_misc_irq_init' [-Werror=missing-prototypes]
  181 | void __init ath79_misc_irq_init(void __iomem *regs, int irq,
      |             ^~~~~~~~~~~~~~~~~~~

Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
---
 drivers/irqchip/irq-ath79-misc.c | 19 +------------------
 1 file changed, 1 insertion(+), 18 deletions(-)

--- a/drivers/irqchip/irq-ath79-misc.c
+++ b/drivers/irqchip/irq-ath79-misc.c
@@ -10,6 +10,7 @@
  *  Parts of this file are based on Atheros' 2.6.15/2.6.31 BSP
  */
 
+#include <asm/time.h>
 #include <linux/irqchip.h>
 #include <linux/irqchip/chained_irq.h>
 #include <linux/of_address.h>
@@ -177,21 +178,3 @@ static int __init ar7240_misc_intc_of_in
 
 IRQCHIP_DECLARE(ar7240_misc_intc, "qca,ar7240-misc-intc",
 		ar7240_misc_intc_of_init);
-
-void __init ath79_misc_irq_init(void __iomem *regs, int irq,
-				int irq_base, bool is_ar71xx)
-{
-	struct irq_domain *domain;
-
-	if (is_ar71xx)
-		ath79_misc_irq_chip.irq_mask_ack = ar71xx_misc_irq_mask;
-	else
-		ath79_misc_irq_chip.irq_ack = ar724x_misc_irq_ack;
-
-	domain = irq_domain_add_legacy(NULL, ATH79_MISC_IRQ_COUNT,
-			irq_base, 0, &misc_irq_domain_ops, regs);
-	if (!domain)
-		panic("Failed to create MISC irqdomain");
-
-	ath79_misc_intc_domain_init(domain, irq);
-}
